zproject/backends.py

```diff
@@ -2802,6 +2876,89 @@ def validate_otp_params(
         raise JsonableError(_("Can't use both mobile_flow_otp and desktop_flow_otp together."))
 
 
+class SAMLSPInitiatedLogout:
+    @classmethod
+    def get_logged_in_user_idp(cls, request: HttpRequest) -> Optional[str]:
+        """
+        Information about the authentication method which was used for
+        this session is stored in authentication_method session attribute.
+        If SAML was used, this extracts the IdP name and returns it.
+        """
+        # Some asserts to ensure this doesn't get called incorrectly:
+        assert hasattr(request, "user")
+        assert isinstance(request.user, UserProfile)
+
+        authentication_method = request.session.get("authentication_method", "")
```

Comment:
So I know that sessions created before this upgrade won't have this parameter, and that's OK. But are there other cases where an authenticated session might not have one? 

I guess one possibility on my mind is whether logging in via JWT or via an invitation sets this, and if so, to what value? I could imagine `ZulipDummyBackend` being what you end up with? I don't think that causes an obvious problem for how this PR uses the session parameters, but that might be worth adjusting in a follow-up so that we track it as a JWT login, for example.

Reply:
Yeah probably a different name like `social_auth_backend` would be better here.